application-analytics-ui/src/main/resources/Analytics/Code/Macros/RowEvolution.xml

```diff
@@ -0,0 +1,289 @@
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require.config({
+  paths: {
+    'chartjs': 'https://cdn.jsdelivr.net/npm/chart.js/dist/Chart.min',
+  }
+});
+
+require(['jquery', 'chartjs'], ($) =&gt; {
+  const PERIODS = ["Day", "Week", "Month", "Year"];
+  let chart;
+
+  /**
+   * Creates the chart
+   *
+   * @param data json with the dataset for the current page.
+   */
+  const renderChart = (data) =&gt; {
+    const dates = data.map(item =&gt; item.date);
+    const field = data.map(item =&gt; item[$('#field').val()] || 0);
+    const ctx = $('#chart-display');
+    if (chart) {
+      chart.destroy();
+    }
+    chart = new Chart(ctx, {
+      type: 'line',
+      data: {
+        labels: dates,
+        datasets: [{
+          label: '',
+          data: field,
+          borderColor: 'rgba(75, 192, 192, 1)',
+          backgroundColor: 'rgba(75, 192, 192, 0.2)',
+          fill: false
+        }]
+      },
+      options: {
+        scales: {
+          y: {
+            beginAtZero: true
+          }
+        }
+      }
+    });
+    $('#chartModal').modal('show');
+  };
+
+  /**
+   * Fetches the data from Matomo using the XWiki server as a proxy through the RowEvolutionJSON page. It then saves the data in session 
+   * storage for easy access. This allows the user to display a different field for the same time interval without making a new request and updates the chart.
+   *
+   * @param parameter parameter need for the request
+   */
+  const fetchChartData = (parameter) =&gt; {
+    $('#options').data('parameter', parameter);
+    const options = $('#options')
+    const period = options.data('period')
+    const date = options.data('date');
+    const requestPage = options.data('requestPage')
+    $.getJSON(`${requestPage}?parameter=${parameter}&amp;period=${period}&amp;date=last${date}`)
```

Comment:
You are doing a request to `RowEvolutionMostViewedPagesJSON?parameter=...` , which isn't an actual URL, so it is resolved to `http://localhost:8080/xwiki/bin/view/MyPage/RowEvolutionMostViewedPagesJSON?parameter=...`.
You probably tested in the right space and didn't noticed